### PR TITLE
connection information alert added when connection not possible

### PR DESCRIPTION
**Pull request by tomeks86**

### CODE REVIEW

File: client/src/main/java/com/battleships/start_window/ConnectionFailedDialogController.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package com.battleships.start_window;
```
Comment (totine) [inline client/src/main/java/com/battleships/start_window/ConnectionFailedDialogController.java]: self comment: unnecessary

File: client/src/main/java/com/battleships/start_window/data_insertion/ConnectionSettingsPaneController.java [hunk 2]
```diff
@@ -96,10 +112,19 @@ private boolean isPortAndIPPresent() {
     }
 
     private Optional<String> getOptionalIPIfInsertedCorrectly() {
-        return Optional.of(ipTextField.textProperty().get().split(":")[0]);
+        String ip = ipTextField.textProperty().get().split(":")[0].trim();
```
Comment (totine) [inline client/src/main/java/com/battleships/start_window/data_insertion/ConnectionSettingsPaneController.java]: extract to method

File: client/src/main/java/com/battleships/start_window/data_insertion/ConnectionSettingsPaneController.java [hunk 3]
```diff
@@ -96,10 +112,19 @@ private boolean isPortAndIPPresent() {
...
+            return Optional.of(ipTextField.textProperty().get().split(":")[0]);
+        } else {
+            return Optional.empty();
+        }
     }
 
     private Optional<Integer> extractPortIfPlayerInserted() {
-        return Optional.of(Integer.valueOf(ipTextField.getText().split(":")[1]));
+        try {
+            return Optional.of(Integer.valueOf(ipTextField.getText().split(":")[1]));
```
Comment (totine) [inline client/src/main/java/com/battleships/start_window/data_insertion/ConnectionSettingsPaneController.java]: as above

**Review by totine — CHANGES_REQUESTED** [review]
Minor fixes to do

**Review by jeger — APPROVED** [review]

**Review by totine — APPROVED** [review]

File: client/src/main/java/com/battleships/start_window/data_insertion/ConnectionSettingsPaneController.java [hunk 4]
```diff
@@ -96,10 +109,27 @@ private boolean isPortAndIPPresent() {
     }
 
     private Optional<String> getOptionalIPIfInsertedCorrectly() {
-        return Optional.of(ipTextField.textProperty().get().split(":")[0]);
+        String ip = extractIPFromIPTextFieldContent();
+        if (!ip.isEmpty()) {
```
Comment (makoJohn) [inline client/src/main/java/com/battleships/start_window/data_insertion/ConnectionSettingsPaneController.java:113]: Optional.of(ip).filter(s -> !s.isEmpty());

**Pull request merged**